sklearn/utils/metadata_requests.py

```diff
@@ -10,8 +10,15 @@ class RequestType(Enum):
     UNREQUESTED = False
     REQUESTED = True
     ERROR_IF_PASSED = None
+    # this sentinel is used in `_metadata_request__*` attributes to indicate that a
+    # metadata is not present eventhough it may be present in the corresponding
+    # method's signature.
+    # TODO: I'm not happy with REMOVE, for now putting a Yoda style name here.
+    EXISTS_NOT = sentinel("EXISTS_NOT")
```

Comment:
Please check this is picklable.

Reply:
Being picklable is one of the things mentioned in the PEP and is in the test suite:

https://github.com/taleinat/python-stdlib-sentinels/blob/79f8cf748541d379250f2d98ed008b89695d90f0/test/test_sentinels.py#L42-L43